homeassistant/components/overkiz/cover.py

```diff
@@ -1,40 +1,413 @@
 """Support for Overkiz covers - shutters etc."""
-from pyoverkiz.enums import OverkizCommand, UIClass
+from __future__ import annotations
 
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import Any, cast
+
+from pyoverkiz.enums import (
+    OverkizCommand,
+    OverkizCommandParam,
+    OverkizState,
+    UIClass,
+    UIWidget,
+)
+from pyoverkiz.models import Device
+
+from homeassistant.components.cover import (
+    ATTR_POSITION,
+    ATTR_TILT_POSITION,
+    CoverDeviceClass,
+    CoverEntity,
+    CoverEntityDescription,
+    CoverEntityFeature,
+)
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import Platform
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityDescription
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from . import HomeAssistantOverkizData
 from .const import DOMAIN
-from .cover_entities.awning import Awning
-from .cover_entities.generic_cover import OverkizGenericCover
-from .cover_entities.vertical_cover import LowSpeedCover, VerticalCover
+from .coordinator import OverkizDataUpdateCoordinator
+from .entity import OverkizDescriptiveEntity
+
+
+def is_closed(device: Device) -> bool | None:
+    """Return if the cover is closed."""
+
+    if state := device.states[OverkizState.CORE_OPEN_CLOSED]:
+        return state.value == OverkizCommandParam.CLOSED
```

Comment:
The previous code in Generic cover was a lot more complex than that.  
Maybe it was covering edge cases?